ValheimPlus/Inventory.cs

```diff
@@ -1,19 +1,132 @@
-﻿using HarmonyLib;
-using System;
-using ValheimPlus.Configurations;
-
-namespace ValheimPlus
-{
-    [HarmonyPatch(typeof(Inventory), "IsTeleportable")]
-    public static class noItemTeleportPrevention
-    {
-        private static void Postfix(ref Boolean __result)
-        {
-            if (Configuration.Current.Items.IsEnabled)
-            {
-                if (Configuration.Current.Items.noTeleportPrevention)
-                    __result = true;
-            }
-        }
-    }
-}
+﻿using HarmonyLib;
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+using ValheimPlus.Configurations;
+
+namespace ValheimPlus
+{
+    [HarmonyPatch(typeof(Inventory), "IsTeleportable")]
+    public static class noItemTeleportPrevention
+    {
+        private static void Postfix(ref Boolean __result)
+        {
+            if (Configuration.Current.Items.IsEnabled)
+            {
+                if (Configuration.Current.Items.noTeleportPrevention)
+                    __result = true;
+            }
+        }
+    }
+
+    [HarmonyPatch(typeof(Inventory), "TopFirst", new Type[] { typeof(ItemDrop.ItemData) })]
+    public static class changeInventoryFillOrder
+    {
+        public static bool Prefix(ref Boolean __result)
+        {
+            if (Configuration.Current.Inventory.IsEnabled &&
+                Configuration.Current.Inventory.inventoryFillTopToBottom)
+            {
+                __result = true;
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+    }
+
+    [HarmonyPatch(typeof(Inventory), MethodType.Constructor)]
+    [HarmonyPatch(new Type[] { typeof(string), typeof(Sprite), typeof(int), typeof(int) })]
+    public static class changeInventorySlotCount
+    {
+        static void Prefix(ref string name, ref Sprite bkg, ref int w, ref int h)
+        {
+            if (Configuration.Current.Inventory.IsEnabled)
+            {
+                // Player inventory number of rows, clamp value in range 4-20
+                int playerInventoryRows = Math.Min(20, Math.Max(4, Configuration.Current.Inventory.playerInventoryRows));

```

Comment:
let's move these clamps to consts on the class

Reply:
I cannot do that as i get a [C0133](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0133) error
I could not have them at all and just assign them to h and w. ie: h = Math.....

Reply:
> let's move these clamps to consts on the class

I cannot do that as i get a [C0133](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0133) error
I could not have them at all and just assign them to h and w. ie: h = Math.....

Reply:
oh sorry I mean the '20', '4', values set to consts like:
```
private const int playerInventoryMaxRows = 20;
private const int playerInventoryMinRows = 4;
```

Reply:
ah, makes sense, done